neptune/new/internal/backends/hosted_client.py

```diff
@@ -0,0 +1,134 @@
+        "headers": {"X-Neptune-LegacyClient": "false"}
+    }
+}
+
+
+def create_http_client(ssl_verify: bool, proxies: Dict[str, str]) -> RequestsClient:
+    http_client = RequestsClient(ssl_verify=ssl_verify)
+    http_client.session.verify = ssl_verify
+
+    update_session_proxies(http_client.session, proxies)
+
+    return http_client
+
+
+def _get_token_client(credentials: Credentials, ssl_verify: bool, proxies: Dict[str, str]):
+    token_http_client = create_http_client(ssl_verify, proxies)
+
+    config_api_url = credentials.api_url_opt or credentials.token_origin_address
+    if proxies is None:
+        verify_host_resolution(config_api_url)
+
+    return create_swagger_client(
+        build_operation_url(config_api_url, BACKEND_SWAGGER_PATH),
+        token_http_client
+    )
+
+
+@lru_cache(maxsize=None, typed=True)
+def get_client_config(credentials: Credentials, ssl_verify: bool, proxies: Dict[str, str]) -> ClientConfig:
+    backend_client = _get_token_client(credentials=credentials, ssl_verify=ssl_verify, proxies=proxies)
+
+    config = backend_client.api.getClientConfig(
+        X_Neptune_Api_Token=credentials.api_token,
+        alpha="true",
+        **DEFAULT_REQUEST_KWARGS,
+    ).response().result
+
+    if hasattr(config, "pyLibVersions"):
+        min_recommended = getattr(config.pyLibVersions, "minRecommendedVersion", None)
+        min_compatible = getattr(config.pyLibVersions, "minCompatibleVersion", None)
+        max_compatible = getattr(config.pyLibVersions, "maxCompatibleVersion", None)
+    else:
+        raise UnsupportedClientVersion(neptune_client_version, max_version="0.4.111")
+
+    return ClientConfig(
+        api_url=config.apiUrl,
+        display_url=config.applicationUrl,
+        min_recommended_version=version.parse(min_recommended) if min_recommended else None,
+        min_compatible_version=version.parse(min_compatible) if min_compatible else None,
+        max_compatible_version=version.parse(max_compatible) if max_compatible else None
+    )
+
+
+@lru_cache(maxsize=None, typed=True)
+def create_http_client_with_auth(credentials: Credentials, ssl_verify: bool, proxies: Dict[str, str]):
+    token_client = _get_token_client(credentials=credentials, ssl_verify=ssl_verify, proxies=proxies)
+
+    http_client = create_http_client(ssl_verify=ssl_verify, proxies=proxies)
+    http_client.authenticator = NeptuneAuthenticator(
+        credentials.api_token,
+        token_client,
+        ssl_verify,
+        proxies
+    )
+    user_agent = 'neptune-client/{lib_version} ({system}, python {python_version})'.format(
+        lib_version=neptune_client_version,
+        system=platform.platform(),
+        python_version=platform.python_version())
+    http_client.session.headers.update({'User-Agent': user_agent})
+
+    return http_client
+
+
+@lru_cache(maxsize=None, typed=True)
+def create_backend_client(client_config: ClientConfig, http_client: HttpClient) -> SwaggerClient:
```

Comment:
one more thing @Raalsky : I've merged current master to `dev/mgmt`, could you please rebase and apply the same treatment to artifact client?

Reply:
Sute.